index.js

```diff
@@ -17,105 +23,145 @@ const urlencodedParser = bodyParser.urlencoded({ extended: false })
+		socket_obj.join(room_id)
+		room.addPlayer(socket_obj.id)
+		result = room
+	}
+	// drops connection if not validated
+	else {
+		socket_obj.disconnect(true)
+	}
+	return result
 }
 
 // there exists documentation for Socket IO Middleware
 // possible to implement before io.on connection
-io.on('connection', socket => {
-    // validate all incoming connections
-    const room = validateSocket(socket)
-    // stops server-side operations on invalid socket connection
-    if (room == null) return
+io.on("connection", socket => {
+	// validate all incoming connections
+	const room = validateSocket(socket)
+	// stops server-side operations on invalid socket connection
+	if (room == null) return
+	// dynamically assign host
+	if (socket.id == room.listOfPlayerSockets[0]) {
+		socket.emit("assigned_as_host")
+	}
+
+	socket.emit("room_id", room.id)
+	socket.emit("player_number", room.getPlayerNumberFromSocketID(socket.id))
+	io.to(room.id).emit("player_list", room.listOfPlayerSockets)
 
-    socket.emit('room_id', room.id)
-    socket.emit('player_number', room.getPlayerNumberFromSocketID(socket.id))
-    // needed to enable turn for room creator
-    io.to(room.getCurrentPlayerSocket()).emit('enable_turn', true)
+	// check room after player disconnect
+	socket.on("disconnect", () => {
+		const playerNumberInRoom = room.getPlayerNumberFromSocketID(socket.id)
+		room.removePlayer(socket.id)
+		// removes room from memory if empty
+		if (room.checkIfEmpty()) {
+			setOfActiveGames.remove(room.id)
+		}
+		// alert players a client left
+		else {
+			io.in(room.id).emit("player_list", room.listOfPlayerSockets)
+			// move room to next player if needed
+			if (playerNumberInRoom == room.currentPlayerNumber) {
+				room.incrementPlayerNumber()
+				const currPlayer = room.getCurrentPlayerSocket()
+				io.in(room.id).emit("current_player", currPlayer)
+				io.to(currPlayer).emit("enable_turn")
+				console.log(currPlayer)
+			}
+		}
+	})
 
-    // update users in room for new image
-    socket.on('gameplay_stroke', image_as_json => {
-        // send all clients in room new image
-        io.in(room.id).emit('new_image', image_as_json)
-        room.nextTurn()
-        // has to reperform operation due to new room state
-        io.to(room.getCurrentPlayerSocket()).emit('enable_turn', true)
-    });
-});
+	// update users in room for new image
+	socket.on("gameplay_stroke", image_as_json => {
+		// send all clients in room new image
+		io.in(room.id).emit("new_image", image_as_json)
+		// execute room logic
+		room.nextTurn()
+		// alert clients in room to move to next turn
+		const currPlayer = room.getCurrentPlayerSocket()
+		io.in(room.id).emit("current_player", currPlayer)
+		io.to(currPlayer).emit("enable_turn")
+	})
+
+	socket.on("request_game_start", () => {
+		// do start game logic
+		room.startGame()
+		io.to(room.fakeArtist).emit("assigned_fake_artist")
+		io.in(room.id).emit("start_game", room.clue)
```

Comment:
any way you can not emit the clue to fake artist as well ?

Reply:
yes, but i dont think its worth the processing since the client won't have access to the socket obj anyways
our option would be to loop through the connections in each rook and emit if not fake